pkg/kratos/handlers.go

```diff
@@ -753,6 +785,29 @@ func (a *API) handleUpdateSettingsFlow(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// populate returnTo url if not set
+	if flow.ReturnTo == nil {
+		if continueWith, ok := flow.GetContinueWithOk(); ok {
+			for _, c := range continueWith {
+				if continueRedirect := c.ContinueWithRedirectBrowserTo; continueRedirect != nil {
+					flow.SetReturnTo(continueRedirect.RedirectBrowserTo)
+					break
+				}
+			}
+		}
+	}
```

Comment:
Just to be clear, are we re-writing the flow object only for the redirect in the `if` block right after? If yes, wouldn't something like this be clearer:
```
returnTo = flow.ReturnTo or flow.ContinueWithRedirectBrowserTo
redirectUserTo(returnTo)
```

This way we don't have to edit the flow object